amazon/ion/ioncmodule.c

```diff
@@ -86,6 +86,9 @@ static PyObject* digits_str;
 static PyObject* fractional_precision_str;
 static PyObject* store_str;
 
+// Note: We cast x (assumed to be an int) to a long so that we can cast to a pointer without warning.
+#define INT_TO_ION_TYPE(x) ( (ION_TYPE)(long)(x) )
```

Comment:
Added this macro to handle converting the int versions of `ION_TYPE` back into the pointer type. For some reason converting an int to a pointer is considered sus, but a long to a pointer is A-OK.